### PR TITLE
Meteor v2.6 and projections option support

### LINKED ISSUE #386: Meteor v2.6 support (mongo@1.14.0)
`fields` field was deprecated in [Meteor v2.6](https://docs.meteor.com/changelog.html#v2620220201) and we need to use `projection` instead.
`redis-oplog` [still uses](https://github.com/cult-of-coders/redis-oplog/blob/b2be29ddc2feee460d248a7e5daffd2883c4e512/lib/cache/ObservableCollection.js#L311) this field, which is now `undefined` and it causes a lot of bugs in my project.

### PR DESCRIPTION
**Pull request by maxpain**
closes #386

### CODE REVIEW

**Comment by plb666** [comment]
I have the same problem in my project, the publication does not consider the fields attribute and publishes all fields. The behavior also poses a security problem since all user fields are sent to the client. As a workaround I override the _getFindOptions method of Mongo Collection:

```
  Mongo.Collection.prototype._getFindOptions = ((func) => {
        return function (options) {
            const findOptions = func.call(this, options);
            if (findOptions.projection) {
                findOptions.fields = findOptions.projection;
            }
            return findOptions;
        };
    })(Mongo.Collection.prototype._getFindOptions);
```

Is there a possibility that this merge request will be added to the package? Thank you.

**Comment by maxpain** [comment]
@theodorDiaconu, could you please look into this?

**Comment by theodorDiaconu** [comment]
@maxpain looks ok, @StorytellerCZ is maintainer

File: lib/cache/ObservableCollection.js [hunk 1]
```diff
@@ -89,34 +90,37 @@ export default class ObservableCollection {
...
-        if (this.options.fields) {
-            this.fieldsArray = _.keys(this.options.fields);
+        if (fields) {
+            this.fieldsArray = _.keys(fields);
 
             if (!_.isArray(this.fieldsArray)) {
                 throw new Meteor.Error(
-                    'We could not properly extract any fields. "fields" must be an object. This was provided: ' +
-                        JSON.stringify(this.options.fields)
+                    'We could not properly extract any fields. "projdction" or "fields" must be an object. This was provided: ' +
```
Comment (StorytellerCZ) [inline lib/cache/ObservableCollection.js:106]: Typo here in the word `projection`

File: lib/cache/ObservableCollection.js [hunk 2]
```diff
@@ -153,7 +157,7 @@ export default class ObservableCollection {
         if (this.matcher) {
             return !!this.collection.findOne(
                 _.extend({}, this.selector, { _id }),
-                { fields: { _id: 1 } }
+                { projection: { _id: 1 } }
```
Comment (StorytellerCZ) [inline lib/cache/ObservableCollection.js:160]: How is the backward compatibility on this one?

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]

File: lib/cache/ObservableCollection.js (hunk 2 above)
Comment (maxpain) [reply lib/cache/ObservableCollection.js]: Ok, let's use `fields` then.

Comment (StorytellerCZ) [reply lib/cache/ObservableCollection.js]: Well, that would be not fix the problem that this PR is trying to fix. I'm just trying to figure out how to have both available here, depending on which one is supported.

Comment (maxpain) [reply lib/cache/ObservableCollection.js]: Meteor will support 'fields' for some time, so we can still use it.

Comment (StorytellerCZ) [reply lib/cache/ObservableCollection.js]: Alright! I will release a patch version with these.

**Pull request merged**